support-frontend/assets/pages/[countryGroupId]/thank-you.tsx

```diff
@@ -12,121 +13,133 @@ export type ThankYouProps = {
 };
 
 export function ThankYou({ geoId, appConfig }: ThankYouProps) {
-	/** 👇 a lot of this is copy/pasted from the checkout */
 	const countryId = CountryHelper.detect();
-
 	const { currencyKey, countryGroupId } = getGeoIdConfig(geoId);
-	const searchParams = new URLSearchParams(window.location.search);
 
-	/** Get and validate product */
+	const searchParams = new URLSearchParams(window.location.search);
+	// product
 	const productParam = searchParams.get('product');
 	const productKey =
 		productParam && isProductKey(productParam) ? productParam : undefined;
 	const product = productKey && productCatalog[productKey];
-	if (!product) {
-		return <div>Product not found</div>;
-	}
-
-	/** Get and validate ratePlan */
+	// ratePlan
 	const ratePlanParam = searchParams.get('ratePlan');
 	const ratePlanKey =
-		ratePlanParam && ratePlanParam in product.ratePlans
+		ratePlanParam && product && ratePlanParam in product.ratePlans
 			? ratePlanParam
 			: undefined;
-	const ratePlan = ratePlanKey && product.ratePlans[ratePlanKey];
-	if (!ratePlan) {
-		return <div>Rate plan not found</div>;
-	}
-
-	/** Get and validate the amount */
+	const ratePlan =
+		ratePlanKey && product ? product.ratePlans[ratePlanKey] : undefined;
+	// contributionAmount
 	const contributionParam = searchParams.get('contribution');
 	const contributionAmount = contributionParam
 		? parseInt(contributionParam, 10)
 		: undefined;
-	let promotion;
+
 	let payment: {
 		originalAmount: number;
 		discountedAmount?: number;
 		contributionAmount?: number;
 		finalAmount: number;
 	};
+	let promotion: Promotion | undefined;
 
-	if (productKey === 'Contribution') {
-		if (!contributionAmount) {
-			return <div>Contribution not specified</div>;
-		}
-
+	if (!productParam && !ratePlanParam) {
```

Comment:
This component has a lot of logic again, good job getting it into one component it might not be obvious it caters for one time

Reply:
On the plus side,  removing `oneTimeThankyou.tsx` simplifies the routing (bit more to go I know) and we now have less repetition of code checking the `contribution` ParamUrl..